realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -173,9 +160,25 @@ private void emitAccessors(JavaWriter writer) throws IOException {
                 writer.emitStatement(
                         "realm.checkIfValid()"
                 );
+
+                // For String and bytes[], null value will be returned by JNI code. Try to save one JNI call here.
+                if (metadata.isNullable(field) && !Utils.isString(field) && !Utils.isByteArray(field)) {
+                    writer.beginControlFlow("if (row.isNull(%s))", staticFieldIndexVarName(field));
```

Comment:
Isn't `row.isNull` also a JNI call? So doing `row.isNull()` and `row.getString(foo)` the same speedwise?